openvino/util/include/openvino/util/file_util.hpp

```diff
@@ -0,0 +1,248 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <fstream>
+#include <functional>
+#include <string>
+#include <vector>
+
+#ifndef ENABLE_UNICODE_PATH_SUPPORT
+#    ifdef _WIN32
+#        if defined __INTEL_COMPILER || defined _MSC_VER
+#            define ENABLE_UNICODE_PATH_SUPPORT
+#        endif
+#    elif defined(__GNUC__) && (__GNUC__ > 5 || (__GNUC__ == 5 && __GNUC_MINOR__ > 2)) || defined(__clang__)
+#        define ENABLE_UNICODE_PATH_SUPPORT
+#    endif
+#endif
+
+namespace ov {
+namespace util {
+
+/// OS specific file traits
+template <class C>
+struct FileTraits;
+
+template <>
+struct FileTraits<char> {
+    static constexpr const auto file_separator =
+#ifdef _WIN32
+        '\\';
+#else
+        '/';
+#endif
+    static constexpr const auto dot_symbol = '.';
+    static std::string library_ext() {
+#ifdef _WIN32
+        return {"dll"};
+#else
+        return {"so"};
+#endif
+    }
+    static std::string library_prefix() {
+#ifdef _WIN32
+        return {""};
+#else
+        return {"lib"};
+#endif
+    }
+};
```

Comment:
this was taken from IE, can we remove IE stuff and re-use this one there?
E.i. we remove IE's implementation and use this one. 
For header's only code we can use `using ov::utils::FileUtils`

Reply:
Removed